### DISCUSSION TITLE
Use local LSP binary instead of Mason-manged one

### DISCUSSION THREAD

**Discussion by nounder** [LazyVim/LazyVim#4201] [General]
For some scenarios it might be necessary to use local LSP binaries (found in virtualenv, node_modules, or nix env) rather than global ones managed by Mason.

I tried setting `mason = false ` without any effect. LSP command still points to Mason-managed binary.
```lua
return {
  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      servers = {
        denols = {
          mason = false,
        },
      },
    },
  },
}
```

Setting `enabled = false` cause unintended consequence in JS/TS environment due to [vtsls/denols handling](https://github.com/LazyVim/LazyVim/blob/12818a6cb499456f4903c5d8e68af43753ebc869/lua/lazyvim/plugins/lsp/init.lua#L248-L257).

Is there a way to disable auto-install for specific LSPs in Mason or at least change priority how LSP binary is found in PATH?

**Comment by dpetka2001** [comment]
Just uninstall the LSP server you want from Mason UI and it should not install again?

**Reply by nounder** [reply]
Indeed, uninstalling LSP worked after disabling Mason in LSP server worked. Apparently the option is only checked during server installation/check phase; not when setting up a workspace.

**Reply by dpetka2001** [reply]
Mason doesn't have a mechanic for automatically uninstalling servers and by default it prepends its $PATH to your own shell's $PATH. That's why Neovim prefers Mason's binaries first (although it offers a setting to change this if I'm non mistaken?). You can only uninstall the servers yourself if you don't want them.

**Reply by nounder** [reply]
Thanks for clarification. Yes, Mason [seems to support](https://github.com/williamboman/mason.nvim/blob/e2f7f9044ec30067bc11800a9e266664b88cda22/README.md?plain=1#L203-L209) prepending, appending, and even skipping altogether, to the $PATH env var via PATH option.
